app/components/elements/Filters.jsx

```diff
@@ -17,53 +16,69 @@ class Filters extends React.Component {
     });
   }
 
-  handleGenderClick(gender) {
-    this.setFilter("gender", gender.id);
+  handleInputChange = (event) => {
+    this.setFilter("query", event.target.value);
   }
 
-  handleCareerClick(career, checked) {
-    const { careers } = this.props.filters;
-    this.setFilter("careers", checked ? careers.concat([career.id]) : careers.filter((id) => id !== career.id));
-  }
+  handleFilterClick(type, resource, checked) {
+    const selected = this.props.filters[type];
 
-  handleInputChange(event) {
-    this.setFilter("query", event.target.value);
+    this.setFilter(type, checked ? selected.concat([resource.id]) : selected.filter(id => id !== resource.id));
   }
 
   render() {
+    const { careers, traits } = this.props.filters;
+
     return (
       <form className="filters pure-form" onSubmit={e => e.preventDefault()}>
+
+        <h1 className="content-subhead">Search</h1>
+        <fieldset>
+          <input
+            type="text"
+            className="pure-input"
+            value={this.props.filters.query}
+            onChange={this.handleInputChange}
+            placeholder="Search for name, bio, etc"
+          />
+        </fieldset>
+
+        <h1 className="content-subhead">Personal Traits</h1>
+        <fieldset>
+          {this.context.meta.traits.map(trait => (
+            <div className="filter" key={trait.id}>
+              <label htmlFor={`filter-trait-${trait.id}`} className="pure-checkbox">
+                {trait.description}
+              </label>
+              <input
+                id={`filter-trait-${trait.id}`}
+                type="checkbox"
+                value={trait.id}
+                checked={(traits || []).find(id => id === trait.id) ? true : false}
+                onChange={e => this.handleFilterClick("traits", trait, e.target.checked)}
+              />
+            </div>
+          ))}
+        </fieldset>
+
         <h1 className="content-subhead">Career Orientation</h1>
         <fieldset>
           {this.context.meta.careers.map((career, n) => (
             <div className="filter" key={n}>
-            <label
-              htmlFor={"filter-career-" + career.id}
-              className="pure-checkbox"
-            >
-              {career.description} {}
-            </label>
+              <label htmlFor={`filter-career-${career.id}`} className="pure-checkbox">
+                {career.description}
+              </label>
               <input
-                id={"filter-career-" + career.id}
+                id={`filter-career-${career.id}`}
                 type="checkbox"
                 value={career.id}
-                checked={(this.props.filters.careers ||  []).find((id) => id === career.id) ? true : false}
-                onChange={(e) => this.handleCareerClick(career, e.target.checked)}
+                checked={(careers || []).find(id => id === career.id) ? true : false}
```

Comment:
Same as above. We're initializing `this.props.filters.careers` to `[]`, if I'm not mistaken, so we may not need the `|| []`.